decentraland/explorer
# feat: Builder In-World Publish Flow

https://www.figma.com/file/jwcjNS7tibLm1Qbwx4LKif/Builder-in-world?node-id=732%3A2

![image](https://user-images.githubusercontent.com/41490935/113690183-6ca1c280-96cb-11eb-84e2-341db10bfe6d.png)

![image](https://user-images.githubusercontent.com/41490935/113690255-7f1bfc00-96cb-11eb-96f9-94a072492974.png)

![image](https://user-images.githubusercontent.com/41490935/113690298-893dfa80-96cb-11eb-971b-ca3e6c998673.png)

### Notes
- **PROGRESS BAR:** As we curently don't have any way to know the progress of the publish process, we are going to display a fake progress in the bar. This will be temporal until we implement the way of return the publish progress from the kernel side.
- **CANCEL UPLOAD:** This feature has not been implemented because it is something that is not currently implemented on the kernel side. This will be temporal until we implement the way of cancel a publication in the middle of the process from kernel.

### Testing
1. Go to: https://explorer.decentraland.zone/branch/feat/Builder-In-World-Publish-Flow/index.html?ENV=zone&ENABLE_BUILDER_IN_WORLD&position=100,100&realm=fenrir-amber
2. Activate the Builder In World editor pressing the "**K**" key.
3. Click on the publish button.
4. See the new flow.
